test/unit/test_server.py

```diff
@@ -54,7 +54,16 @@ def assert_compatible(name, spec1, spec2):
         assert nargs2 - ndefs2 <= nargs1 - ndefs1, \
             'Incompatible number of non-default args: %r' % name
 
-        assert spec1.args == spec2.args[:nargs1], \
+        # The `policy` arg used to be named  `policy_index` or `policy_idx`
+        # in Swift 2.1 and 2.2. It changed in Swift 2.3.
+        # We rename the arg here to be keep compatibility and have the same
+        # method signature excepted for the name of the `policy` argument.
+        spec1_args = [arg.replace('policy_idx', 'policy').replace(
```

Comment:
Consider putting this in a `if name in ['get_diskfile', 'async_update']:` block.